pkg/network/server.go

```diff
@@ -571,6 +576,9 @@ func (s *Server) handleGetDataCmd(p Peer, inv *payload.Inventory) error {
 			}
 		}
 	}
+	for _, invGroup := range payload.CreateGroup(inv.Type, notFound) {
```

Comment:
You don't really need it here, GetData is limited with `MaxHashesCount`, so the number of missing hashes can't be more than that. But it'd be nice to enforce that restriction in `Inventory` payload decoder.